LeetCode/vDimple/String/StringPermutations_D/StringPermutations.cpp

```diff
@@ -0,0 +1,65 @@
+#include"StringPermutations.h"
+#include<map>
+
+void getStringPermutations(std::string const& str, std::vector<char>& charArr, std::vector<size_t>& charCount, size_t level, std::string& resStr, std::vector<std::string>& result)
+{
+	try
+	{
+		if (level >= str.length())
+		{
+			result.push_back(resStr);
+		}
+		else
+		{
+			for (size_t i = 0; i < charArr.size(); i++)
+			{
+				if (charCount[i] > 0)
+				{
+					resStr[level] = charArr[i];
+					++level;
+					--charCount[i];
+					getStringPermutations(str, charArr, charCount, level, resStr, result);
+					--level;
+					++charCount[i];
+				}
+			}
+		}
+	}
+	catch (...)
+	{
+		throw;
+	}
+}
+
+STRINGPERMUTATIONSD_API std::vector<std::string> getStringPermutations(std::string const& str)
+{
+	try
+	{
+		if (str.length() < 2)
+			throw new std::exception ("\nPlease enter more than 1 character to see its permutations.\n");
+			
+		std::map<char, int> sMap;
```

Comment:
std::map<char, int> 

should we make the map like std::map<char,size_t> ?

Reply:
Incorporated this review comment.